admin/user.go

```diff
@@ -129,43 +129,39 @@ func (s *Service) CreateOrganizationForUser(ctx context.Context, userID, orgName
 	return org, nil
 }
 
-func (s *Service) InviteUserToOrganization(ctx context.Context, email, inviterID, orgID, roleID, orgName, roleName string) error {
-	// Validate email address
-	_, err := mail.ParseAddress(email)
-	if err != nil {
-		return fmt.Errorf("invalid user email address %q", email)
+func (s *Service) InviteUserToOrganization(ctx context.Context, opts *database.InviteUserToOrganizationOptions) error {
+	if err := database.Validate(opts); err != nil {
+		return err
 	}
 
 	// Create invite
-	err = s.DB.InsertOrganizationInvite(ctx, email, orgID, roleID, inviterID)
+	err := s.DB.InsertOrganizationInvite(ctx, opts.Email, opts.OrgID, opts.RoleID, opts.InviterID)
```

Comment:
Sorry, I think I misrepresented in another comment. The options types in the `database` package should directly match the database functions, e.g. `InsertOrganizationInviteOptions`. 

I would probably keep the previous func signature on the `Service` here. The call to `s.Email...` will already validate the email, so just need it to happen on the `InsertOrganizationInvite` call